src/musicbrainz/chromaprinter.cpp

```diff
@@ -12,6 +12,16 @@
 
 namespace
 {
+// Type declarations of *fprint and *encoded pointers need to account for Chromaprint API version
+// (void* -> uint32_t*) and (void* -> char*) changed in versions v1.4.0 or later -- alyptik 12/2016
+#if (CHROMAPRINT_VERSION_MINOR > 3) || (CHROMAPRINT_VERSION_MAJOR > 1)
+    typedef uint32_t* uint32_p;
+    typedef char* char_p;
```

Comment:
I was under the impression that the cpp directives should always be far left. Is it preferred to indent them here? I am fixing the other sections right now.